packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/OutputLibrary.cs

```diff
@@ -10,58 +10,50 @@ namespace Microsoft.Generator.CSharp
 {
     public class OutputLibrary
     {
-        private IReadOnlyList<EnumProvider>? _enums;
-        private IReadOnlyList<ModelProvider>? _models;
         private IReadOnlyList<ClientProvider>? _clients;
 
         public OutputLibrary()
         {
-            EnumMappings = new Dictionary<InputEnumType, EnumProvider>();
-            ModelMappings = new Dictionary<InputModelType, ModelProvider>();
-
             _allModels = new(InitializeAllModels);
         }
 
-        private readonly Lazy<(EnumProvider[] Enums, ModelProvider[] Models)> _allModels;
-
-        private (EnumProvider[] Enums, ModelProvider[] Models) InitializeAllModels()
+        private (TypeProvider[] Enums, TypeProvider[] Models) InitializeAllModels()
         {
             var input = CodeModelPlugin.Instance.InputLibrary.InputNamespace;
 
-            var enums = new EnumProvider[input.Enums.Count];
+            var enums = new TypeProvider[input.Enums.Count];
```

Comment:
Lets just move the enum portion to BuildEnums and the model portion to BuildModels and drop the tuple.